hatchways/team-scallops
# CI/CD testing and automated deployment

### Build Automation:
   - Production build automation & Docker build automation
   - unit testing
   - push to private self-hosted Docker registry
   - deploy to Kubernetes staging server
   - Ability to promote builds from staging to prod.
   - saves test logs at [https://drone.host.destin.dev/destintech/team-scallops](url)


### Any information needed to test this feature (required):
- Update push to drone_ci or staging.
- wait up to 2 minutes max for the cron job on the build server
-  check your build logs [https://drone.host.destin.dev/destintech/team-scallops/](url) 
- navigate to [https://lovingsitter-staging.destin.dev/](url)
- the update will be live once the CD pipeline is complete. 
- Note: The tests are non-blocking, so they are just a log at the moment to inform us all and to track if the build fails :) 
 

### Any issues with the current functionality (optional):
- Testing should be blocking, but it would cause strain to the other developers. I've implemented a better alternative, to display the test results so we can keep each other up to date on the progress and manually check the build with a badge from the CICD server. 
### Badge example:
[![Build Status](https://drone.host.destin.dev/api/badges/destintech/team-scallops/status.svg?ref=staging)](https://drone.host.destin.dev/destintech/team-scallops)
